tests/gis_tests/gis_migrations/test_operations.py

```diff
@@ -82,9 +82,9 @@ def alter_gis_model(self, migration_class, model_name, field_name,
         operation = migration_class(*args)
         new_state = project_state.clone()
         operation.state_forwards('gis', new_state)
+        self.current_state = new_state
```

Comment:
Did you run into problems that necessitated this change? It seems like it would cause incorrect tear down if `operation.database_forwards` fails (trying to rollback that operation that failed).